taipy/core/data/_abstract_sql.py

```diff
@@ -226,9 +226,14 @@ def _read_as_pandas_dataframe(
         join_operator=JoinOperator.AND,
     ):
         with self._get_engine().connect() as conn:
+            result = conn.execute(text(self._get_read_query(operators, join_operator)))
+
+            # On pandas 1.3.5 there's a bug that makes that the dataframe from sqlalchemy query is
+            # created without headers
+            keys = [col for col in result.keys()]
```

Comment:
Isn't that enough?
```suggestion
            keys = result.keys()
```
Or:
```suggestion
            keys = list(result.keys())
```

Reply:
Unfortunately result.keys returns an sqlalchemy object that does support list casting, thats why I've used a list comprehension.